index.js

```diff
@@ -205,36 +208,110 @@ Hook.prototype.initialize = function initialize() {
 };
 
 /**
+ * Stashes unstaged changes.
+ *
+ * @param {Function} done Callback
+ * @api private
+ */
+Hook.prototype._stash = function stash(done) {
+  var hooked = this;
+
+  spawn(hooked.git, ['stash', '--keep-index', '--include-untracked'], {
+    env: process.env,
+    cwd: hooked.root,
+    stdio: [0, 1, 2]
+  }).once('close', function() {
+    // a nonzero here may be that there are no unstaged changes.
+    done();
+  });
+};
+
+/**
+ * Unstashes changes ostensibly stashed by {@link Hook#_stash}.
+ *
+ * @param {Function} done Callback
+ * @api private
+ */
+Hook.prototype._unstash = function unstash(done) {
+  var hooked = this;
+
+  spawn(hooked.git, ['stash', 'pop'], {
+    env: process.env,
+    cwd: hooked.root,
+    stdio: [0, 1, 2]
+  }).once('close', function(code) {
```

Comment:
You might as well supply the `done` method here directly to eliminate this function call.